app/cdap/components/DataPrep/TopPanel/PipelineConfigHelper.js

```diff
@@ -14,682 +14,86 @@
-
-const PREFIX = 'features.DataPrep.PipelineError';
+import VersionStore from 'services/VersionStore';
 
 export default function getPipelineConfig() {
-  let workspaceInfo = DataPrepStore.getState().dataprep.workspaceInfo;
-
-  return MyDataPrepApi.getInfo().mergeMap((res) => {
-    if (res.statusCode === 404) {
-      console.log(`can't find method; use latest wrangler-transform`);
-      return constructProperties(workspaceInfo);
-    }
-
-    let pluginVersion = res.values[0]['plugin.version'];
-    return constructProperties(workspaceInfo, pluginVersion);
-  });
-}
-
-function findWranglerArtifacts(artifacts, pluginVersion) {
-  let wranglerArtifacts = artifacts.filter((artifact) => {
-    if (pluginVersion) {
-      return artifact.name === 'wrangler-transform' && artifact.version === pluginVersion;
-    }
-
-    return artifact.name === 'wrangler-transform';
-  });
-
-  if (wranglerArtifacts.length === 0) {
-    // cannot find plugin. Error out
-    throw T.translate(`${PREFIX}.missingWranglerPlugin`);
-  }
-
-  let filteredArtifacts = wranglerArtifacts;
-
-  if (!pluginVersion) {
-    let highestVersion = findHighestVersion(
-      wranglerArtifacts.map((artifact) => {
-        return artifact.version;
-      }),
-      true
-    );
-
-    filteredArtifacts = wranglerArtifacts.filter((artifact) => {
-      return artifact.version === highestVersion;
-    });
-  }
-
-  let returnArtifact = filteredArtifacts[0];
-
-  if (filteredArtifacts.length > 1) {
-    returnArtifact.scope = SCOPES.USER;
-  }
-
-  return returnArtifact;
-}
-
-function constructFileSource(artifactsList, properties) {
-  if (!properties) {
-    return null;
-  }
-
-  let plugin = objectQuery(properties, 'values', '0');
-
-  let pluginName = Object.keys(plugin)[0];
+  const workspaceId = DataPrepStore.getState().dataprep.workspaceId;
 
-  plugin = plugin[pluginName];
-  let batchArtifact = find(artifactsList, { name: 'core-plugins' });
-  let realtimeArtifact = find(artifactsList, { name: 'spark-plugins' });
-  if (!batchArtifact) {
-    return T.translate(`${PREFIX}.fileBatch`);
-  }
-
-  if (!realtimeArtifact) {
-    return T.translate(`${PREFIX}.fileRealtime`);
-  }
-
-  batchArtifact.version = '[1.7.0, 3.0.0)';
-  realtimeArtifact.version = '[1.7.0, 3.0.0)';
-
-  let batchPluginInfo = {
-    name: plugin.name,
-    label: plugin.name,
-    type: 'batchsource',
-    artifact: batchArtifact,
-    properties: plugin.properties,
+  const params = {
+    context: getCurrentNamespace(),
+    workspaceId,
   };
 
-  let realtimePluginInfo = Object.assign({}, batchPluginInfo, {
-    type: 'streamingsource',
-    artifact: realtimeArtifact,
-  });
+  return MyDataPrepApi.getSpecification(params).map((res) => {
+    const batchArtifact = {
+      name: 'cdap-data-pipeline',
+      version: VersionStore.getState().version,
+      scope: SCOPES.SYSTEM,
+    };
 
-  let batchStage = {
-    name: 'File',
-    plugin: batchPluginInfo,
-  };
+    // TODO: Realtime artifact
```

Comment:
Is there a Jira for this?

Reply:
https://cdap.atlassian.net/browse/CDAP-18067